File: api_call.rb
require 'rubygems'
require 'dogapi'
require 'pry'
require 'dotenv/load'

api_key = ENV['DATADOG_API_KEY']
app_key = ENV['DATADOG_APP_KEY']

dog = Dogapi::Client.new(api_key, app_key)

title = 'My_Metric_API'
description = 'Doing the my_metric through the API'
graphs = [{
    "definition" => {
        "events" => [],
        "requests" => [{
            "q" => "anomalies(avg:postgresql.rows_fetched{*}, 'basic', 2)",
            "type" => "line"
        }],
        "viz" => "timeseries"
    },
    "title" => "Rows Fetched With Anomoly Function"
},
{
    "definition" => {
        "events" => [],
        "requests" => [{
            "q" => "max:my_metric{host:Alekss-MBP.fios-router.home}",
            "type" => "bars"
        }],
        "viz" => "timeseries"
    },
    "title" => "My Metric API Timeseries"
},
{
    "definition" => {
        "events" => [],
        "requests" => [{
            "q" => "sum:my_metric{host:Alekss-MBP.fios-router.home}.rollup(sum, 60)",
            "aggregator": "sum"
        }],
        "viz" => "query_value",
        "precision": 0,
        "autoscale": false
    },
    "title" => "My Metric With Rollup Function"
}]
template_variables = [{
    "name" => "host1",
    "prefix" => "host",
    "default" => "host:Alekss-MBP.fios-router.home"
}]

dog.create_dashboard(title, description, graphs, template_variables)

File: code/api_call.py
from datadog import initialize, api

options = {
    'api_key': 'xxxxxxx',
    'app_key': 'xxxxxxx'
}

initialize(**options)

title = 'My Custom Timeboard'
widgets = [
{
    'definition': {
        'type': 'timeseries',
        'requests': [
            {'q': 'avg:my_metric{host:ec2.datadog}'}
        ],
        'title': 'my_metric scoped over host'
}},
     {
      'definition': {
        'type': 'timeseries',
        'requests': [
            {'q': "anomalies(avg:mysql.performance.user_time{*}, 'robust', 2)"}
        ],
        'title': 'MySQL metric with Anamoly function'
}},
{
    'definition': {
        'type': 'timeseries',
        'requests': [
            {'q': 'avg:my_metric{host:ec2.datadog}.rollup(sum, 3600)'}
        ],
        'title': 'my_metric with rollup for last hour'
}}
]

layout_type = 'ordered'
description = 'A dashboard with various metrics'
#is_read_only = True
#notify_list = ['user@domain.com']
#template_variables = [{
#    'name': 'host1',
#    'prefix': 'host',
#    'default': 'my-host'
#}]

api.Dashboard.create(title=title,
                     widgets=widgets,
                     layout_type=layout_type,
                     description=description,
#                     is_read_only=is_read_only,
#                     notify_list=notify_list,
#                     template_variables=template_variables
)
